linera-core/src/remote_node.rs

```diff
@@ -238,4 +239,51 @@ impl<N: ValidatorNode> RemoteNode<N> {
         }
         Ok(hashes)
     }
+
+    #[instrument(level = "trace", skip(validators))]
+    async fn download_blob(validators: &[Self], blob_id: BlobId) -> Option<Blob> {
+        // Sequentially try each validator in random order.
+        let mut validators = validators.iter().collect::<Vec<_>>();
+        validators.shuffle(&mut rand::thread_rng());
+        for remote_node in validators {
+            if let Some(blob) = remote_node.try_download_blob(blob_id).await {
+                return Some(blob);
+            }
+        }
+        None
+    }
+
+    #[instrument(level = "trace", skip(validators))]
+    pub async fn download_certificate_for_blob_from_validators_futures(
```

Comment:
No comment? I thought we should have docs on pub methods :)

Reply:
(I only moved this in this PR; but you're right, I'll add a comment. Also, this doesn't even need to be `async`!)

Reply:
(Actually, I'll inline it.)

Reply:
We also don't need to shuffle the validators.

I'll merge this PR, and then make this change in a separate one.